File: backend/app/schemas/gestion_directiva.py
from pydantic import BaseModel


class PlaneacionCursoResponse(BaseModel):
    id: int
    periodo: str
    curso: str
    cupos_solicitados: int
    grupos_estimados: int
    cupo_grupo: int
    grupos_aprobados: int

    class Config:
        from_attributes = True


class PlaneacionCursoStatsResponse(BaseModel):
    total_cursos: int
    total_cupos_solicitados: int
    total_grupos_estimados: int
    total_grupos_aprobados: int
    promedio_cupo_grupo: float


class ProgramacionCursoResponse(BaseModel):
    id: int
    periodo: str
    curso: str
    grupo: str
    profesor: str | None
    vinculacion: str | None
    horario: str | None
    edificio: str | None
    salon: str | None
    matriculados: int
    cancelaron: int
    aprobaron: int
    reprobaron: int

    class Config:
        from_attributes = True


class ProgramacionCursoStatsResponse(BaseModel):
    total_grupos: int
    total_matriculados: int
    total_cancelaron: int
    total_aprobaron: int
    total_reprobaron: int
    tasa_aprobacion: float
    tasa_cancelacion: float


class ApoyoEconomicoResponse(BaseModel):
    id: int
    periodo: str
    nivel: str | None
    tipo_evento: str | None
    lugar: str | None
    pais: str | None
    participantes: int
    rol: str | None
    apoyo_economico: float

    class Config:
        from_attributes = True


class ApoyoEconomicoStatsResponse(BaseModel):
    total_eventos: int
    total_participantes: int
    total_apoyo_economico: float
    promedio_apoyo_economico: float

File: frontend/src/models/GestionDirectiva.ts
export interface PlaneacionCurso {
  id: number;
  periodo: string;
  curso: string;
  cupos_solicitados: number;
  grupos_estimados: number;
  cupo_grupo: number;
  grupos_aprobados: number;
}

export interface ProgramacionCurso {
  id: number;
  periodo: string;
  curso: string;
  grupo: string;
  profesor: string | null;
  vinculacion: string | null;
  horario: string | null;
  edificio: string | null;
  salon: string | null;
  matriculados: number;
  cancelaron: number;
  aprobaron: number;
  reprobaron: number;
}

export interface ApoyoEconomico {
  id: number;
  periodo: string;
  nivel: string | null;
  tipo_evento: string | null;
  lugar: string | null;
  pais: string | null;
  participantes: number;
  rol: string | null;
  apoyo_economico: number;
}

export interface PlaneacionStats {
  total_cursos: number;
  total_cupos_solicitados: number;
  total_grupos_estimados: number;
  total_grupos_aprobados: number;
  promedio_cupo_grupo: number;
}

export interface ProgramacionStats {
  total_grupos: number;
  total_matriculados: number;
  total_cancelaron: number;
  total_aprobaron: number;
  total_reprobaron: number;
  tasa_aprobacion: number;
  tasa_cancelacion: number;
}

export interface ApoyoStats {
  total_eventos: number;
  total_participantes: number;
  total_apoyo_economico: number;
  promedio_apoyo_economico: number;
}

// Chart data types
export interface DemandChartData {
  curso: string;
  cupos_solicitados: number;
  matriculados: number;
}

export interface ParticipationChartData {
  name: string;
  asistentes: number;
  cancelados: number;
}

export interface VinculacionChartData {
  name: string;
  value: number;
}

export interface CancelacionChartData {
  horario: string;
  cancelados: number;
}

export interface ApoyoChartData {
  tipo_evento: string;
  monto: number;
}
